src/main/java/com/gw2auth/oauth2/server/service/application/client/authorization/ApplicationClientAuthorizationServiceImpl.java

```diff
@@ -151,6 +159,15 @@ public void save(OAuth2Authorization authorization) {
 
         final Map<String, Object> attributes = new HashMap<>(authorization.getAttributes());
         final Set<String> authorizedScopes = authorization.getAuthorizedScopes();
+        final Set<String> effectiveAuthorizedScopes = Optional.ofNullable(authorizedScopes)
+                .map((scopes) -> {
```

Comment:
Nit: Unnecessary parenthesis

```suggestion
                .map(scopes -> {
```

Reply:
personal preference :)